requirements.txt

```diff
@@ -19,4 +19,10 @@ albumentations>=1.3.0
 pytube>=12.1.3
 pyffmpeg==2.0
 GitPython>=3.1.18
+pytorch-metric-learning==1.6.3
+faiss-cpu==1.7.4
+hawk-eyes==2.1.2
+onnx==1.14.1
+onnxruntime-gpu==1.10.0
```

Comment:
I believe the generic requirements.txt needs onnruntime (1.16.3), and then the requirements-cuda* can override this using onnxruntime-gpu?
Have you tested on a machine without GPU?

Reply:
Hawkeyes  requires onnxruntime-gpu==1.10, and onnxruntime-gpu==1.10 requirements are as in below.
![image](https://github.com/MaximIntegratedAI/ai8x-training/assets/80602263/40d3e232-2b74-4c95-9edb-b1ff0c20365d)

The pytorch 1.8 version that we use utilizes cuda 11.1 and cudnn 8005. Therefore, we were getting the following warning and proceeding with the CPU anyway. As this is required only for gt generation on dataset and still faster than the other available models that work on torch-GPU, I decided to proceed with this approach.

> [W:onnxruntime:Default,` onnxruntime_pybind_state.cc:535 CreateExecutionProviderInstance] Failed to create CUDAExecutionProvider. Please reference https://onnxruntime.ai/docs/reference/execution-providers/CUDA-ExecutionProvider.html#requirements to ensure all dependencies are met.

As the hawk-eyes specifically require onnxruntime-gpu==1.10, and onnxruntime-gpu includes both CUDAExecutionProvider and CPUExecutionProvider, I believe we won't need to install onnxruntime separately for only CPU cases. Furthermore, with Pytorch 2.1 version this code should work automatically on GPU, and if user uses the torch-CPU version, onnxruntime-gpu should switch to CPUExecutionProvider.

Reply:
You can filter warnings, for example this filters a "Profiler function xyz will be ignored" message from torch._dynamo.variables.torch:
```
        pattern = re.compile('.*Profiler function .* will be ignored')
        logging.getLogger('torch._dynamo.variables.torch').addFilter(
            lambda record: not pattern.search(record.getMessage())
        )
```

Reply:
It seems like onnxruntime uses a separate logger. Suppressed onnxruntime warnings by setting "onnxruntime.set_default_logger_severity(3)".